src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -1,12 +1,79 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
+import copy
 
-from . import ApiController, RESTController
+from . import ApiController, RESTController, Task
 from .. import mgr
-from ..security import Scope
+from ..exceptions import DashboardException
+from ..security import Permission, Scope
+from ..services.orchestrator import OrchClient
+from ..services.exception import handle_orchestrator_error
+
+
+def host_task(name, metadata, wait_for=10.0):
+    return Task("host/{}".format(name), metadata, wait_for)
+
+
+def merge_with_orchestrator(mgr_hosts):
+    """Merge mgr hosts with orchestrator hosts.
+
+    :param mgr_hosts: hosts returned from mgr
+    :return list of dict with host information
+    """
+    all_hosts = copy.copy(mgr_hosts)
+    orch = OrchClient()
+    if orch.available():
+        mgr_hostnames = {host['hostname'] for host in mgr_hosts}
+        orch_hosts = (dict(hostname=host.name, ceph_version='', services=[])
+                      for host in orch.hosts.list()
+                      if host.name not in mgr_hostnames)
+        all_hosts.extend(orch_hosts)
+    return all_hosts
 
 
 @ApiController('/host', Scope.HOSTS)
 class Host(RESTController):
+
     def list(self):
-        return mgr.list_servers()
+        hosts = mgr.list_servers()
+        if self._has_permissions(Permission.READ, Scope.ORCHESTRATOR):
+            return merge_with_orchestrator(hosts)
```

Comment:
We're not breaking existing clients. I'm :+1: with merging the hosts in here and keep the code as it is.

Reply:
>    2. If we don't like the idea of issuing 2 requests, let's do the math in the back-end but at least let's make it explicit (`/host?sources=[ceph,orchestrator]`)

I prefer not adding a new endpoint and enhancing `/api/host`. How about adding a new field `sources` in each host object:

```javascript
[
  {
    "hostname": "host1",
    "sources": {
      "ceph": true,
      "orchestrator": true
    }
  },
  {
    "hostname": "host2",
    "sources": {
      "ceph": false,
      "orchestrator": true
    }
  }
]
```

By default, a GET to `/api/host` returns all hosts included orchestrator ones. And then implement a filter with `sources` parameter.

Reply:
address in https://github.com/ceph/ceph/pull/29127/commits/cd101da73a4e933100ae3b3dad242c38bedcb4c1

Reply:
Thanks @bk201 , I'm not too fan about extending Resources via query params (that's why I suggested first the new endpoint approach), but I think it's better than to  backward-incompatibly modify the behavior. The reasoning for me not excited about the query param option is because query params work best when they are left just to representation modifiers (pagination, filtering, field selection, sorting, etc.), and are all the same to all endpoints.